lua/nvim-tree/explorer/watch.lua

```diff
@@ -46,39 +32,39 @@ local function is_folder_ignored(path)
   return false
 end
 
-function M.refresh_path(path)
-  log.line("watcher", "node event executing '%s'", path)
-  local n = utils.get_node_from_path(path)
-  if not n then
-    return
+function M.create_watcher(node)
+  if not M.enabled or type(node) ~= "table" then
+    return nil
   end
 
-  local node = utils.get_parent_of_group(n)
-  local project_root, project = reload_and_get_git_project(path)
-  require("nvim-tree.explorer.reload").reload(node, project)
-  update_parent_statuses(node, project, project_root)
-
-  require("nvim-tree.renderer").draw()
-end
-
-function M.create_watcher(absolute_path)
-  if not M.enabled then
-    return nil
+  local path
+  if node.type == "link" then
+    path = node.link_to
+  else
+    path = node.absolute_path
   end
-  if is_git(absolute_path) or is_folder_ignored(absolute_path) then
+
+  if is_git(path) or is_folder_ignored(path) then
     return nil
   end
 
   local function callback(watcher)
-    log.line("watcher", "node event scheduled %s", watcher.context)
+    log.line("watcher", "node event scheduled refresh %s", watcher.context)
     utils.debounce(watcher.context, M.debounce_delay, function()
-      M.refresh_path(watcher._path)
+      if node.link_to then
+        log.line("watcher", "node event executing refresh '%s' -> '%s'", node.link_to, node.absolute_path)
+      else
+        log.line("watcher", "node event executing refresh '%s'", node.absolute_path)
+      end
+      require("nvim-tree.explorer.reload").refresh_node(node)
```

Comment:
This will still result in one refresh for each instance of the symlinked dir, however optimising that away requires significant complexity.